samples/boards/google_pda/src/view.c

```diff
@@ -0,0 +1,348 @@
+
+	if (atomic_test_bit(&vo->flags, FLAGS_SNOOP0)) {
+		smf_set_state(SMF_CTX(vo), &view_states[SNOOP0]);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_SNOOP1)) {
+		smf_set_state(SMF_CTX(vo), &view_states[SNOOP1]);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_SNOOP3)) {
+		smf_set_state(SMF_CTX(vo), &view_states[SNOOP3]);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_NO_CONNECTION)) {
+		if (vo->toggle) {
+			set_led(LED_BLUE);
+			vo->toggle = false;
+		} else {
+			set_led(LED_OFF);
+			vo->toggle = true;
+		}
+	} else if (atomic_test_bit(&vo->flags, FLAGS_CC1_CONNECTION)) {
+		set_led(LED_RED);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_CC2_CONNECTION)) {
+		set_led(LED_GREEN);
+	}
+}
+
+static void snoop2_exit(void *o)
+{
+	set_led(LED_OFF);
+}
+
+static void snoop3_entry(void *o)
+{
+	struct view_obj_t *vo = (struct view_obj_t *)o;
+
+	vo->timeout = K_MSEC(250);
+}
+
+static void snoop3_run(void *o)
+{
+	struct view_obj_t *vo = (struct view_obj_t *)o;
+
+	if (atomic_test_bit(&vo->flags, FLAGS_SNOOP0)) {
+		smf_set_state(SMF_CTX(vo), &view_states[SNOOP0]);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_SNOOP1)) {
+		smf_set_state(SMF_CTX(vo), &view_states[SNOOP1]);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_SNOOP2)) {
+		smf_set_state(SMF_CTX(vo), &view_states[SNOOP2]);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_NO_CONNECTION)) {
+		if (vo->toggle) {
+			set_led(LED_BLUE);
+			vo->toggle = false;
+		} else {
+			set_led(LED_OFF);
+			vo->toggle = true;
+		}
+	} else if (atomic_test_bit(&vo->flags, FLAGS_CC1_CONNECTION)) {
+		set_led(LED_GREEN);
+	} else if (atomic_test_bit(&vo->flags, FLAGS_CC2_CONNECTION)) {
+		set_led(LED_GREEN);
+	}
+}
+
+static void snoop3_exit(void *o)
+{
+	set_led(LED_OFF);
+}
+
+static const struct smf_state view_states[] = {
+	[SNOOP0] = SMF_CREATE_STATE(snoop0_entry, snoop0_run, snoop0_exit),
+	[SNOOP1] = SMF_CREATE_STATE(snoop1_entry, snoop1_run, snoop1_exit),
+	[SNOOP2] = SMF_CREATE_STATE(snoop2_entry, snoop2_run, snoop2_exit),
+	[SNOOP3] = SMF_CREATE_STATE(snoop3_entry, snoop3_run, snoop3_exit),
+};
+
+int view_init(void)
+{
+	int ret;
+
+	if (!device_is_ready(led0.port)) {
+		return -EIO;
+	}
+
+	if (!device_is_ready(led1.port)) {
+		return -EIO;
+
+	}
+
+	if (!device_is_ready(led2.port)) {
+		return -EIO;
+	}
+
+	ret = gpio_pin_configure_dt(&led0, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_configure_dt(&led1, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_configure_dt(&led2, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		return ret;
+	}
+
+	view_obj.flags = ATOMIC_INIT(0);
+
+        return 0;
+}
+
+void main(void)
+{
+	const struct device *dev_pd_analyzer = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
+
+	if (usb_enable(NULL)) {
+		return;
+	}
+
+	meas_init();
+	controls_init();
+	model_init(dev_pd_analyzer);
+	view_init();
+
+	smf_set_initial(SMF_CTX(&view_obj), &view_states[SNOOP0]);
+
+        while(1) {
```

Comment:
see the style guide, space is always required after `while`.

`clang-format` can fix this automatically for you.